File: Sample Pages/sample.php
Test Page (at least vulnerable to xss - don't use on live)<br/>
<?php
echo "<br/>GET input0:<br/>";
echo "<br/>input0=".($_GET["input0"])."<br/>";
echo "strlen(input0)=".strlen($_GET["input0"])."<br/>";
echo "mb_strlen(input0)=".mb_strlen($_GET["input0"], '8bit')."<br/><br/>";

echo "<br/>POST input1:<br/>";
echo "<br/>input1=".($_POST["input1"])."<br/>";
echo "strlen(input1)=".strlen($_POST["input1"])."<br/>";
echo "mb_strlen(input1)=".mb_strlen($_POST["input1"], '8bit')."<br/><br/>";

echo "<br/>COOKIE input2:<br/>";
echo "<br/>input2=".($_COOKIE["input2"])."<br/>";
echo "strlen(input2)=".strlen($_COOKIE["input2"])."<br/>";
echo "mb_strlen(input2)=".mb_strlen($_COOKIE["input2"], '8bit')."<br/><br/>";
?>
<?php
parse_str(file_get_contents("php://input"), $_POST_RAW);

$req_dump=str_repeat("-=", 20)."\r\n\$_SERVER:\r\n".print_r($_SERVER,true)."\r\n\$_POST:\r\n".print_r($_POST,true)."\r\n\$_GET:\r\n".print_r($_GET,true)."\r\n\$_FILES:\r\n".print_r($_FILES,true)."\r\n\$_POST_RAW:\r\n".print_r($_POST_RAW,true)."\r\n\$_COOKIE:\r\n".print_r($_COOKIE,true)."\r\n";

echo "<pre>".$req_dump."</pre>";

echo "<br/>generic parameter ( \$_REQUEST[\"input\"] ) input:<br/>";
echo "<br/>input=".($_REQUEST["input"])."<br/>";
echo "strlen(input)=".strlen($_REQUEST["input"])."<br/>";
echo "mb_strlen(input)=".mb_strlen($_REQUEST["input"], '8bit')."<br/><br/>"; 
?>

File: Sample Pages/sample.py
from django.http import HttpResponse
import re
import os

def index(request):
	result = "Test Page (at least vulnerable to xss - don't use on live)<br/>\n"
	try:
		result += "<br/>GET input0:<br/><br/>"
		result += "input0="+request.GET['input0']+"<br/>"
		result += "len(input0)="+str(len(request.GET['input0']))+"<br/>"
		result += "s.encode('utf-8')(input0)="+str(utf8len(request.GET['input0']))+"<br/><br/>"
	except:
		pass

	try:    
		result += "<br/>POST input1:<br/><br/>"
		result += "input1="+request.POST['input1']+"<br/>"
		result += "len(input1)="+str(len(request.POST['input1']))+"<br/>"
		result += "s.encode('utf-8')(input1)="+str(utf8len(request.POST['input1']))+"<br/><br/>"
	except:
		pass

	try:    
		result += "<br/>COOKIE input2:<br/><br/>"
		result += "input2="+request.COOKIES.get('input2')+"<br/>"
		result += "len(input2)="+str(len(request.COOKIES.get('input2')))+"<br/>"
		result += "s.encode('utf-8')(input2)="+str(utf8len(request.COOKIES.get('input2')))+"<br/><br/>"
	except:
		pass

		
	result += "<br/>\nGET:\n<br/>"
	for key, values in request.GET.lists():
		result += key + "=" + ', '.join(values) + "<br/>\n<br/>"     
	result += "\n<br/>POST:\n<br/>"
	for key, values in request.POST.lists():
		result += key + "=" + ', '.join(values) + "<br/>\n<br/>" 
	result += "\n<br/>HEADERS:\n<br/>"
	#regex_headers = re.compile(r'^(HTTP_.+|CONTENT_TYPE|CONTENT_LENGTH)$')
	#request_headers = {}
	for header in request.META:
		#if regex_headers.match(header):
			result +=   "{header}:{value}<br/>\n<br/>".format(header=header,value=request.META[header])
	
	for key in os.environ.keys():
		result = result + "%30s: %s <br/>\n" % (key,os.environ[key])

	return HttpResponse(result)


def utf8len(s):
  return len(s.encode('utf-8'))